python/cudf/cudf/core/column/categorical.py

```diff
@@ -332,7 +329,7 @@ def children(self):
             codes_column = column.build_column(
                 data=buf,
                 dtype=codes_column.dtype,
-                mask=self.mask,
+                mask=mask_column.mask,
```

Comment:
Do we need to handle the mask at all here? Isn't the `codes` child column guaranteed to be non-null?